src/MCRun.jl

```diff
@@ -260,15 +280,282 @@ function mc_cycle!(mc_states, move_strat, mc_params, pot, ensemble, n_steps, a,
+     write(savefile, "n_atoms: $(mc_params.n_atoms)\n")
+     write(savefile,"n_adjust: $(mc_params.n_adjust)\n")
+
+    #  close(savefile)
+ end
+"""
+    function save_state(savefile::IOStream,mc_state::MCState)
+saves a single mc_state struct to a savefile
+"""
+function save_state(savefile::IOStream,mc_state::MCState)
+    write(savefile,"temp_beta: $(mc_state.temp) $(mc_state.beta) \n")
+    write(savefile,"total_energy: $(mc_state.en_tot)\n")
+    write(savefile,"max_displacement: $(mc_state.max_displ[1]) $(mc_state.max_displ[2]) $(mc_state.max_displ[3])\n")
+    write(savefile, "counts_a/v/r/ex:  $(mc_state.count_atom[1])   $(mc_state.count_atom[2]) $(mc_state.count_vol[1]) $(mc_state.count_vol[2]) $(mc_state.count_rot[1]) $(mc_state.count_rot[2]) $(mc_state.count_exc[1]) $(mc_state.count_exc[2]) \n")
+
+    if length(mc_state.ham) > 2
+        ham1 = sum(mc_state.ham)
+        ham2 = sum( mc_state.ham .* mc_state.ham)
+    elseif length(mc_state.ham) == 2
+        ham1 = mc_state.ham[1]
+        ham2 = mc_state.ham[2]
+    else
+        ham1 = 0
+        ham2 = 0
+    end
+    write(savefile, "E,E2: $ham1 $ham2 \n")
+    if typeof(mc_state.config.bc) == SphericalBC{Float64}
+        write(savefile, "Boundary: $(typeof(mc_state.config.bc))  $(mc_state.config.bc.radius2) \n")
+    elseif typeof(mc_state.config.bc) == PeriodicBC{Float64}
+        write(savefile, "Boundary: $(typeof(mc_state.config.bc))$(mc_state.config.bc.box_length) \n" )
+    end
+    write(savefile,"configuration \n")
+    for row in mc_state.config.pos
+        write(savefile,"$(row[1]) $(row[2]) $(row[3]) \n")
+    end
+
+end
+"""
+    save_results(results::Output; directory = pwd())
+Saves the on the fly results and histogram information for re-reading.
+"""
+function save_results(results::Output; directory = pwd())
+    resultsfile =  open("$(directory)/results.data","w+")
+    rdf_file = open("$directory/RDF.data","w+")
+    write(resultsfile,"emin,emax,nbins= $(results.en_min) $(results.en_max) $(results.n_bin) \n")
+    write(resultsfile, "Histograms \n")
+    writedlm(resultsfile,results.en_histogram)
+    close(resultsfile)
+    writedlm(rdf_file,results.rdf)
+    close(rdf_file)
+    #requires: en_min,en_max,n_bin,en_hist
+    #reading doesn't require the rest as that is handled as a post-process
+end
+"""
+    function save_states(mc_params,mc_states,trial_index; directory = pwd())
+opens a savefile, writes the mc params and states and the trial at which it was run. 
+"""
+function save_states(mc_params,mc_states,trial_index, directory)
+    i = 0 
+    savefile = open("$(directory)/save.data","w+")
+    write(savefile,"Save made at step $trial_index \n") #at $(format(now(),"HH:MM") )\n")
+    save_params(savefile,mc_params)
+    for state in mc_states
+        i += 1
+        write(savefile, "config $i \n")
+        save_state(savefile,state)
+        write(savefile,"end \n")
+    end
+    close(savefile)
+end
+
```

Comment:
The functions ABOVE this line are the save files used to reinitialise MC_Run

Reply:
good. Do we want more specific names?

Reply:
Presently no, eventually this will become a checkpoint function